lib/motors/controller.cpp

```diff
@@ -2,8 +2,142 @@
+      break;
+    case 0xFF0F:
+      logger_.log(core::LogLevel::kFatal, "ERROR_EMERGENCY_BUTTON: Emergency button pressed");
+      break;
+    case 0xFF10:
+      logger_.log(core::LogLevel::kFatal,
+                  "ERROR_CONTROLLER_OVERTEMPERATURE: Controller overtemperature");
+      break;
+    case 0x3210:
+      logger_.log(core::LogLevel::kFatal,
+                  "ERROR_DC_LINK_OVERVOLTAGE: Power supply voltage too high");
+      break;
+    default:
+      logger_.log(core::LogLevel::kFatal,
+                  "GENERIC_ERROR: Unspecific error occurred with code %i",
+                  error_code);
+      break;
+  }
+}
+
+ControllerStatus Controller::processWarningMessage(const std::uint8_t warning_code)
+{
+  // Flag specific warnings in binary
+  if (warning_code & 0x2) {
+    logger_.log(core::LogLevel::kFatal,
+                "Contoller Warning: Motor Temperature Exceeded (code: %x)",
+                warning_code);
+    return ControllerStatus::kUnrecoverableWarning;
+  }
+  if (warning_code & 0x4) {
+    logger_.log(
+      core::LogLevel::kFatal, "Contoller Warning: DC link under voltage (code: %x)", warning_code);
+    return ControllerStatus::kUnrecoverableWarning;
+  }
+  if (warning_code & 0x8) {
+    logger_.log(
+      core::LogLevel::kFatal, "Contoller Warning: DC link over voltage (code: %x)", warning_code);
+    return ControllerStatus::kUnrecoverableWarning;
+  }
+  if (warning_code & 0x10) {
+    logger_.log(core::LogLevel::kFatal,
+                "Contoller Warning: DC over priority_errorent (code: %x)",
+                warning_code);
+    return ControllerStatus::kUnrecoverableWarning;
+  }
+  if (warning_code & 0x20) {
+    logger_.log(core::LogLevel::kFatal,
+                "Contoller Warning: Stall protection active (code: %x)",
+                warning_code);
+    return ControllerStatus::kUnrecoverableWarning;
+  }
+  if (warning_code & 0x40) {
+    logger_.log(
+      core::LogLevel::kFatal, "Contoller Warning: Max velocity exceeded (code: %x)", warning_code);
+    return ControllerStatus::kUnrecoverableWarning;
+  }
+  if (warning_code & 0x80) {
+    logger_.log(
+      core::LogLevel::kFatal, "Contoller Warning: BMS Proposed Power (code: %x)", warning_code);
+    return ControllerStatus::kUnrecoverableWarning;
+  }
+  if (warning_code & 0x100) {
+    logger_.log(core::LogLevel::kFatal,
+                "Contoller Warning: Capacitor temporature exceeded (code: %x)",
+                warning_code);
+    return ControllerStatus::kUnrecoverableWarning;
+  }
+  if (warning_code & 0x200) {
+    logger_.log(
+      core::LogLevel::kFatal, "Contoller Warning: I2T protection (code: %x)", warning_code);
+    return ControllerStatus::kUnrecoverableWarning;
+  }
+  if (warning_code & 0x400) {
+    logger_.log(
+      core::LogLevel::kFatal, "Contoller Warning: Field weakening active (code: %x)", warning_code);
+    return ControllerStatus::kUnrecoverableWarning;
```

Comment:
And I am not at all keen on having to look up error codes in the manual to understand our own logs in every case where there is more than 1 warning active